CodeModel/CodeParsers/CSharp/CaDETClassMetricCalculator.cs

```diff
@@ -37,10 +37,127 @@ internal class CaDETClassMetricCalculator
+            return parsedClass.Members.Where(m => !m.Type.Equals(CaDETMemberType.Property)).Sum(m => m.Metrics[CaDETMetric.CYCLO]);
+        }
+
+        // Public attributes for NOPA metric do not include constants and static fields. (Object-oriented metrics in practice)
+        private static int CountPublicAttributes(CaDETClass parsedClass)
+        {
+            var publicFields = GetPublicFields(parsedClass.Fields);
+            var staticFields = (from field in publicFields
+                             from modifier in field.Modifiers
+                             where modifier.Value.Equals(CaDETModifierValue.Static)
+                             select field).ToList();
+            return publicFields.Where(f => !staticFields.Contains(f)).Count();
+        }
+
+        private static double CountWeightOfClass(CaDETClass parsedClass)
+        {
+            return (double)CountFunctionalPublicMethods(parsedClass.Members) / (CountPublicProperties(parsedClass.Members) + GetPublicFields(parsedClass.Fields).Count());
+        }
+
+        // Public fields for WOC metric do not include constants. (Object-oriented metrics in practice)
+        private static IEnumerable<CaDETField> GetPublicFields(List<CaDETField> fields)
+        {
+            var publicFields = FindFieldsWithModifier(fields, CaDETModifierValue.Public);
+            var constants = FindFieldsWithModifier(publicFields, CaDETModifierValue.Const);
+            return publicFields.Where(f => !constants.Contains(f));
+        }
+
+        private static IEnumerable<CaDETField> FindFieldsWithModifier(IEnumerable<CaDETField> fields, CaDETModifierValue modifierValue)
+        {
+            return (from field in fields
+                    from modifier in field.Modifiers
+                    where modifier.Value.Equals(modifierValue)
+                    select field);
+        }
+
+        private static int CountPublicProperties(IEnumerable<CaDETMember> members)
+        {
+            var properties = members.Where(m => m.Type.Equals(CaDETMemberType.Property));
+            var publicProperties = FindMembersWithModifier(properties, CaDETModifierValue.Public);
+            return publicProperties.Count();
+        }
+
+        // Functional public methods do not include get/set, constructor and abstract methods. (Object-oriented metrics in practice)
+        private static int CountFunctionalPublicMethods(List<CaDETMember> members)
+        {
+            var methods = GetMethods(members);
+            var publicMethods = FindMembersWithModifier(methods, CaDETModifierValue.Public);
+            var abstractMethods = FindMembersWithModifier(publicMethods, CaDETModifierValue.Abstract);
+            return publicMethods.Where(m => !abstractMethods.Contains(m)).Count();
+        }
+
+        private static IEnumerable<CaDETMember> FindMembersWithModifier(IEnumerable<CaDETMember> members, CaDETModifierValue modifierValue)
+        {
+            return (from member in members
```

Comment:
Don't use this syntax. While there is nothing wrong with it, it is a coding style that we do not follow in the rest of the project and we shouldn't mix styles.

If you need a utility function in CaDETMember or CaDETModifier add it.